# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/socket/nsISSLSocketControl.idl
# Commit: 632eefa7e529
# Full Hash: 632eefa7e529e806bdbecfb50e23f33e2dd86e37
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-17 21:50:08
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Add a callback when the handshake is done. r=necko-reviewers,kershaw,keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123601
# ==============================================================================

diff -r fe6b8f41ca83 -r 632eefa7e529 netwerk/socket/nsISSLSocketControl.idl
--- a/netwerk/socket/nsISSLSocketControl.idl	Fri Sep 17 13:19:00 2021 +0000
+++ b/netwerk/socket/nsISSLSocketControl.idl	Fri Sep 17 13:19:00 2021 +0000
@@ -8,6 +8,7 @@
 
 interface nsIInterfaceRequestor;
 interface nsIX509Cert;
+interface nsITlsHandshakeCallbackListener;
 
 %{C++
 #include "nsStringFwd.h"
@@ -171,5 +172,6 @@
      * Disable early data.
      */
     [noscript] void disableEarlyData();
+
+    [noscript] void setHandshakeCallbackListener(in nsITlsHandshakeCallbackListener callback);
 };
-